geomstats/geometry/product_manifold.py

```diff
@@ -172,4 +172,7 @@ def random_uniform(self, n_samples, point_type=None):
         point = [
             space.random_uniform(n_samples)
             for space in self.manifolds]
-        return gs.stack(point, axis=1)
+        samples = gs.stack(point, axis=1)
+        if n_samples == 1 and samples.shape[0] == 1:
```

Comment:
why the second condition? if there is only one sample don't we have `samples.shape[0] = n_manifolds`?

Reply:
Not necessarily, we could have samples.shape = (dim,) since most of the random_uniform(1) functions now squeeze the 0th dimension.

But actually, since the random_uniform functions are properly vectorized now, this condition is not needed at all.